src/main/java/hec/stats/MeanComputable.java

```diff
@@ -0,0 +1,14 @@
+package hec.stats;
+
+public class MeanComputable implements Computable{
+    @Override
+    public float compute(float[] values){
+        //calculate the mean of values
+        float sum = 0f;
+        float[] rval = new float[values.length];
+        for (int i = 0; i <= values.length; i++) {
```

Comment:
this should not be equals - it should be <, because of the zero basis of arrays.

Reply:
Arguably it also shouldn't be a for loop: https://www.baeldung.com/java-8-primitive-streams 

(has a very low code min/max/avg example)

For the parts like what I'm directly replying to the overhead of coverting the List to float[] should be minimal. (Though worth measuring. But I've always thought correctness first was more important that speed first.)

Reply:
I am also using this as an opportunity to teach how to do the computation - we will up to streaming stats. I also am interested in time trials along the way, so we will probably spend some time doing comparisons if you are interested in helping guide that! boxing and unboxing might be an issue when we are dealing with data of the size of time series of ensembles